packages/compass-shell/src/modules/worker-runtime/runtime.js

```diff
@@ -0,0 +1,106 @@
+import path from 'path';
+import fs from 'fs';
+// eslint-disable-next-line camelcase
+import child_process from 'child_process';
+import crypto from 'crypto';
+import electron from 'electron';
+import worker from 'inline-entry-loader!./worker';
+import { createCaller, exposeAll } from './rpc';
+
+function createChildFromSource(src) {
+  const tmpFile = path.resolve(
+    (electron.app || electron.remote.app).getPath('temp'),
+    `worker-runtime-${crypto.randomBytes(32).toString('hex')}`
+  );
+
+  // eslint-disable-next-line no-sync
+  fs.writeFileSync(tmpFile, src, 'utf8');
+
+  const childProcess = child_process.spawn(process.execPath, [tmpFile], {
+    stdio: ['inherit', 'inherit', 'inherit', 'ipc'],
+    env: { ...process.env, ELECTRON_RUN_AS_NODE: 1 },
+    /**
+     * This would be awesome, but only supported since node 12.16.0, compass
+     * is currently at 12.4
+     * @see https://nodejs.org/api/child_process.html#child_process_advanced_serialization
+     */
+    serialization: 'advanced',
+  });
+
+  return childProcess;
+}
```

Comment:
Now that we have our worker as a source string, we can create a child process by just writing in somewhere in the fs in the runtime and running the file we created.

Having an actual file somewhere in fs is a requirement to be able to use `child_procees.spawn` or a `worker_thread` and without this workaround we would need to make a lot of changes around how Compass is bundled as an single-packaged application

Reply:
> Having an actual file somewhere in fs is a requirement to be able to use `child_procees.spawn` or a `worker_thread` and without this workaround we would need to make a lot of changes around how Compass is bundled as an single-packaged application

Thinking about this … you can always just spawn `node` and pipe the script into it on stdin, so maybe that’s actually a better option here? I just didn’t think of that before :smile:

Reply:
🤯 !!! I'll try that, this sounds as an awesome approach!

Reply:
Okay, so I'm sure this will work but I'm a little hesitant to spawn `node` directly as we can't guarantee that Compass users will have it installed, right? And seems like electron runtime (`process.execPath`) just exists immediately if no file path is provided, so I'm not sure if we can pipe a script into it 🤔 Maybe I'm missing something and there is way, maybe you'd have an idea...

Reply:
So, maybe I’m missing something, but running this from the shell in the compass repo seems to work:

```js
$ echo 'console.log("hi")' | env ELECTRON_RUN_AS_NODE=1 electron 
hi
```

Reply:
Yeah, you're right, it works just fine, I messed up the `stdio` option when I tried it and this is why it wasn't working for me. Love the idea of not writing a file so will try to do a full implementation with this approach!

Reply:
Implemented in 37538dc6e398568907a55f3392d3555730381283